pkg/querier/blocks_scanner.go

```diff
@@ -135,9 +140,12 @@ func (d *BlocksScanner) scan(ctx context.Context) error {
 	return nil
 }
 
-func (d *BlocksScanner) scanBucket(ctx context.Context) error {
+func (d *BlocksScanner) scanBucket(ctx context.Context) (err error) {
```

Comment:
Nit: Calling this e.g. “returnErr” would minimize changes and also make it clearer.